lib/node_modules/@stdlib/math/base/special/negafibonacci/src/main.c

```diff
@@ -0,0 +1,121 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/negafibonacci.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/constants/float64/max_safe_nth_fibonacci.h"
+
+static const int64_t negafibonacci_value[ 79 ] = {
+	0,
+	1,
+	-1,
+	2,
+	-3,
+	5,
+	-8,
+	13,
+	-21,
+	34,
+	-55,
+	89,
+	-144,
+	233,
+	-377,
+	610,
+	-987,
+	1597,
+	-2584,
+	4181,
+	-6765,
+	10946,
+	-17711,
+	28657,
+	-46368,
+	75025,
+	-121393,
+	196418,
+	-317811,
+	514229,
+	-832040,
+	1346269,
+	-2178309,
+	3524578,
+	-5702887,
+	9227465,
+	-14930352,
+	24157817,
+	-39088169,
+	63245986,
+	-102334155,
+	165580141,
+	-267914296,
+	433494437,
+	-701408733,
+	1134903170,
+	-1836311903,
+	2971215073,
+	-4807526976,
+	7778742049,
+	-12586269025,
+	20365011074,
+	-32951280099,
+	53316291173,
+	-86267571272,
+	139583862445,
+	-225851433717,
+	365435296162,
+	-591286729879,
+	956722026041,
+	-1548008755920,
+	2504730781961,
+	-4052739537881,
+	6557470319842,
+	-10610209857723,
+	17167680177565,
+	-27777890035288,
+	44945570212853,
+	-72723460248141,
+	117669030460994,
+	-190392490709135,
+	308061521170129,
+	-498454011879264,
+	806515533049393,
+	-1304969544928657,
+	2111485077978050,
+	-3416454622906707,
+	5527939700884757,
+	-8944394323791464
+};
+
+/**
+* Computes the nth negaFibonacci number.
+*
+* @param n    input value
+* @return     output value
+*
+* @example
+* double out = stdlib_base_negafibonacci( 1 );
+* // returns 1
+*/
+double stdlib_base_negafibonacci( const int32_t n ) {
+	int32_t an = stdlib_base_abs( n );
+	if ( n > 0 || an > STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FIBONACCI ) {
+		return 0.0 / 0.0; // NaN
+	}
+	return negafibonacci_value[ an ];
+}
```

Comment:
The approach seems good to me, but question is shall we stick with `js` implementation or this is fine. 
cc @kgryte

Reply:
Is this approach good to go, or should we change it somewhere ? Any suggestions @kgryte ?